File: csharp/channel_partner/InsertChannelPartnerShipToPreference.cs
using System;
using System.Collections.Generic;
using System.Reflection;
using com.ultracart.admin.v2.Api;
using com.ultracart.admin.v2.Model;

namespace SdkSample.channel_partner
{
    public class InsertChannelPartnerShipToPreference
    {
        /*
         Inserts a channel partner shipto preference for a channel partner.
         These preferences are used by EDI channel partners to automatically
         apply return policies and add additional free items to EDI orders based on the EDI code that is present.

         Possible Errors:
         Attempting to interact with a channel partner other than the one tied to your API Key:
            "Invalid channel_partner_oid specified.  Your REST API key may only interact with channel_partner_oid: 12345"
         Supplying a bad channel partner oid: "Invalid channel_partner_oid specified."
        */
        public static void Execute()
        {
            Console.WriteLine("--- " + MethodBase.GetCurrentMethod()?.DeclaringType?.Name + " ---");
            
            try
            {
                // Create channel partner API instance using API key
                ChannelPartnerApi channelPartnerApi = new ChannelPartnerApi(Constants.ChannelPartnerApiKey);
                
                int channelPartnerOid = 12345;
                
                ChannelPartnerShipToPreference preference = new ChannelPartnerShipToPreference();
                preference.ChannelPartnerOid = channelPartnerOid;
                preference.ShipToEdiCode = "EDI_CODE_HERE";
                preference.ReturnPolicy = "This is some return policy text that will be printed on the packing slip.";
                preference.AdditionalKitComponentItemIds = new List<string> { "ITEM_ID1", "ITEM_ID2", "ITEM_ID3" };
                preference.Description = "This is a merchant friendly description to help me remember what the above setting are.";
                
                var apiResponse = channelPartnerApi.InsertChannelPartnerShipToPreference(channelPartnerOid, preference);
                
                if (apiResponse.Error != null)
                {
                    Console.Error.WriteLine(apiResponse.Error.DeveloperMessage);
                    Console.Error.WriteLine(apiResponse.Error.UserMessage);
                    Environment.Exit(1);
                }
                
                var insertedPreference = apiResponse.ShipToPreference;
                
                // This should equal what you submitted.
                Console.WriteLine(insertedPreference);
                Console.WriteLine("Ship to preference inserted successfully");
            }
            catch (Exception ex)
            {
                Console.WriteLine($"Error: {ex.Message}");
                Console.WriteLine(ex.StackTrace);
            }
        }
    }
}

File: python/channel_partner/insert_channel_partner_ship_to_preference.py
from ultracart.apis import ChannelPartnerApi
from ultracart.models import ChannelPartnerShipToPreference
from samples import channel_partner_api_client

# Initialize API
channel_partner_api = ChannelPartnerApi(channel_partner_api_client())
channel_partner_oid = 12345

# Create preference object
preference = ChannelPartnerShipToPreference()
preference.channel_partner_oid = channel_partner_oid
preference.ship_to_edi_code = 'EDI_CODE_HERE'
preference.return_policy = "This is some return policy text that will be printed on the packing slip."
preference.additional_kit_component_item_ids = ['ITEM_ID1', 'ITEM_ID2', 'ITEM_ID3']
preference.description = "This is a merchant friendly description to help me remember what the above setting are."

# Insert the preference
api_response = channel_partner_api.insert_channel_partner_ship_to_preference(channel_partner_oid, preference)

if api_response.error is not None:
    print(api_response.error.developer_message)
    print(api_response.error.user_message)
    exit()

inserted_preference = api_response.ship_to_preference
print(inserted_preference)